Model/Quote/GetCartLineItems.php

```diff
@@ -73,54 +130,111 @@ public function getItems(CartInterface $quote): array
+        return $item->getName();
+    }
+
+    /**
+     * Gets the price of a line item
      *
      * @param CartItemInterface $item
      * @return int
      */
-    private function extractLineItemQuantity(CartItemInterface $item): int
+    private function getLineItemPrice(CartItemInterface $item)
     {
-        $parentItem = $item->getParentItem();
-        if ($parentItem) {
-            $item = $parentItem;
+        $item = $item->getParentItem() ?: $item;
+        return $this->convertToCents((float)$item->getPrice());
+    }
+
+    /**
+     * Gets the weight of a line item in grams
+     *
+     * @param CartItemInterface $item
+     * @return float
+     */
+    private function getLineItemWeightInGrams(CartItemInterface $item): float
+    {
+        $unit = strtolower(
+            $this->scopeConfig->getValue('general/locale/weight_unit', ScopeInterface::SCOPE_STORE)
+        );
+        $weight = $item->getWeight();
+        if ($unit === 'kgs') {
+            return round($weight * 1000, 2);
+        } elseif ($unit === 'lbs') {
+            return round($weight * 453.59237, 2);
         }
 
-        return (int)$item->getQty();
+        return $weight;
     }
 
     /**
-     * Get quote item discount amount.
+     * Gets the line item's image. Falls back to the parent item (If available) if the direct
+     * item does not have an image
      *
      * @param CartItemInterface $item
-     * @return float
+     * @return string
+     */
+    private function getLineItemImage(CartItemInterface $item): string
+    {
+        $product = $this->productRepository->getById($item->getProductId());
+        if ($product->getThumbnail() && $product->getThumbnail() !== 'no_selection') {
+            return $this->productUrlBuilder->getUrl($product->getThumbnail(), 'product_thumbnail_image');
+        }
+        // Attempting to get the parent product if there is one
+        if ($item->getParentItem()) {
+            $image = $this->productRepository->getById($item->getParentItem()->getProductId())->getThumbnail();
+            if ($image) {
+                return $this->productUrlBuilder->getUrl($image, 'product_thumbnail_image');
+            }
+        }
+        return $this->productUrlBuilder->getUrl('no_selection', 'product_thumbnail_image');
+    }
+
+    /**
+     * Get quote item quantity considering product type.
+     *
+     * @param CartItemInterface $item
+     * @return int
      */
-    private function getPriceAdjustment(CartItemInterface $item)
+    private function extractLineItemQuantity(CartItemInterface $item): int
     {
         $parentItem = $item->getParentItem();
-        $childProduct = $item->getProduct();
-        $baseProductPrice = $childProduct->getPrice();
         if ($parentItem) {
             $item = $parentItem;
         }
-        $priceAdjustment = $item->getBasePrice() - $baseProductPrice;
-
-        return $priceAdjustment * 100;
+        return (int)$item->getQty();
```

Comment:
Question: Is there a scenario where the quantity can have a decimal value and we would like to round it instead?

Reply:
I believe you can sell things like fabric that comes in fractional amounts but Checkout doesn't support that